cmd/tnf/addclaim/addclaim.go

```diff
@@ -27,12 +29,194 @@ var (
 		Short: "The test suite generates a \"claim\" file",
 		RunE:  claimUpdate,
 	}
+	claimCompareFiles = &cobra.Command{
+		Use:   "compare",
+		Short: "Compare 2 \"claim\" file",
+		RunE:  claimCompare,
+	}
 )
 
 const (
 	claimFilePermissions = 0o644
 )
 
+type cniplugin struct {
+	name   string
+	plugin interface{}
+}
+type Cni struct {
+	Claim struct {
+		Nodes struct {
+			CniPlugins map[string][]struct {
+				Name    string        `json:"name"`
+				Plugins []interface{} `json:"plugins"`
+			} `json:"cniPlugins"`
+		} `json:"nodes"`
+	} `json:"claim"`
+}
+type Csi struct {
+	Claim struct {
+		Nodes struct {
+			CsiDriver interface{} `json:"csiDriver"`
+		} `json:"nodes"`
+	} `json:"claim"`
+}
+
+type HwInfo struct {
+	Claim struct {
+		Nodes struct {
+			NodesHwInfo map[string]interface{} `json:"nodesHwInfo"`
+		} `json:"nodes"`
+	} `json:"claim"`
+}
+
+type RawResult struct {
+	Claim struct {
+		RawResults struct {
+			Cnfcertificationtest struct {
+				Testsuites struct {
+					Testsuite struct {
+						Testcase []struct {
+							Name   string `json:"-name"`
+							Status string `json:"-status"`
+						} `json:"testcase"`
+					} `json:"testsuite"`
+				} `json:"testsuites"`
+			} `json:"cnf-certification-test"`
+		} `json:"rawResults"`
+	} `json:"claim"`
+}
+
+func claimCompare(cmd *cobra.Command, args []string) error {
+	//var cnis []cniplugin
+	//var plugin interface{}
+	//var name string
+	var nodes, nodes2 []string
+	claimFileTextPtr := &Claim1
+	dat, err := os.ReadFile(*claimFileTextPtr)
+	if err != nil {
+		log.Fatalf("Error reading claim file :%v", err)
+	}
+	claimFileTextPtr2 := &Claim2
+	dat2, err2 := os.ReadFile(*claimFileTextPtr2)
+	if err != nil {
+		log.Fatalf("Error reading claim file2 :%v", err2)
+	}
+	// cniclaimRoot.Claim.RawResults
+	var cni2 Cni
+	err = json.Unmarshal(dat2, &cni2)
+	var cni Cni
+	err = json.Unmarshal(dat, &cni)
+	// csi
+	var csi, csi2 Csi
+	err = json.Unmarshal(dat2, &csi)
+	err = json.Unmarshal(dat2, &csi2)
+	// HwInfo
+	var hwinfo, hwinfo2 HwInfo
+	err = json.Unmarshal(dat, &hwinfo)
+	err = json.Unmarshal(dat2, &hwinfo2)
+	// rawResult
+	var rawResult, rawResult2 RawResult
+	err = json.Unmarshal(dat, &rawResult)
+	err = json.Unmarshal(dat2, &rawResult2)
+	for node, val := range cni.Claim.Nodes.CniPlugins {
+		for node2, val2 := range cni2.Claim.Nodes.CniPlugins {
+			if node == node2 {
+				c, s := compare2cnis(val, val2)
+				log.Info("node ", node2, "has diff plugins ", c, " and cni are not the same ", s)
```

Comment:
log.Info("node ", node2, "has diff plugins ", c, " and cni are not the same ", s)
->
if len(s) !=0 {
  log.Info("node ", node2, "cnis found in claim1 but not present in claim2: ", s)
}
if len (c) !=0{

}